File: src/server/lib/java/org/meet4xmas/wire/Appointment.java
package org.meet4xmas.wire;

import org.meet4xmas.util.StringUtil;

public class Appointment implements java.io.Serializable {
  public int identifier;
  public String creator; // userId
  public int locationType; // use values in Location.LocationType
  public Location location;
  public Participant[] participants;
  public String message;

  public String toString() {
    StringBuilder sb = new StringBuilder("Appointment(");
    sb.append("@identifier: ").append(identifier).append(", ");
    sb.append("@locationType: ").append(Location.LocationType.toString(locationType)).append(", ");
    sb.append("@location: ").append(StringUtil.ValueOrNullToString(location)).append(", ");
    sb.append("@participants: ").append(StringUtil.ArrayOrNullToString(participants, new StringUtil.StringArrayElementVisitor()));
    return sb.toString();
  }
}

File: src/wp7/Meet4Xmas/org.meet4xmas.wire/Appointment.cs
using System.Text;

namespace org.meet4xmas.wire
{
    public partial class Appointment
    {
        public int identifier;
        public string creator; // userId
        public int locationType; // use values in Location.LocationType
        public Location location;
        public Participant[] participants;
        public string message;
        public bool isFinal;

        public string ToString()
        {
            StringBuilder sb = new StringBuilder("<Appointment ");
            sb.Append("@identifier: ").Append(identifier).Append(", ");
            sb.Append("@locationType: ").Append(Location.LocationType.ToString(locationType)).Append(", ");
            sb.Append("@location: ").Append(location).Append(", ");
            sb.Append("@participants: ").Append(participants).Append(", ");
            sb.Append("@isFinal: ").Append(isFinal);
            sb.Append(">");
            return sb.ToString();
        }
    }
}
